crates/nu-command/src/conversions/into/int.rs

```diff
@@ -134,28 +181,120 @@ pub fn action(input: &Value, span: Span) -> Value {
+                }
+            }
+            val.to_string()
+        }
+        _ => {
+            return Value::Error {
+                error: ShellError::UnsupportedInput(
+                    "only strings or integers are supported".to_string(),
+                    head,
+                ),
+            }
+        }
+    };
+    match i64::from_str_radix(&i, radix) {
+        Ok(n) => Value::Int { val: n, span: head },
+        Err(reason) => Value::Error {
+            error: ShellError::CantConvert("".to_string(), reason.to_string(), head),
+        },
+    }
+}
+
 fn int_from_string(a_string: &str, span: Span) -> Result<i64, ShellError> {
-    match a_string.parse::<i64>() {
-        Ok(n) => Ok(n),
-        Err(_) => match a_string.parse::<f64>() {
-            Ok(f) => Ok(f as i64),
-            _ => Err(ShellError::CantConvert(
-                "into int".into(),
-                "string".into(),
-                span,
-            )),
+    let trimmed = a_string.trim();
+    match trimmed {
+        b if b.starts_with("0b") => {
+            let num = match i64::from_str_radix(b.trim_start_matches("0b"), 2) {
+                Ok(n) => n,
+                Err(reason) => {
+                    return Err(ShellError::CantConvert(
+                        "could not parse as integer".to_string(),
+                        reason.to_string(),
+                        span,
+                    ))
+                }
+            };
+            Ok(num)
+        }
+        h if h.starts_with("0x") => {
+            let num = match i64::from_str_radix(h.trim_start_matches("0x"), 16) {
+                Ok(n) => n,
+                Err(reason) => {
+                    return Err(ShellError::CantConvert(
+                        "could not parse as int".to_string(),
+                        reason.to_string(),
+                        span,
+                    ))
+                }
+            };
+            Ok(num)
+        }
+        _ => match a_string.parse::<i64>() {
+            Ok(n) => Ok(n),
+            Err(_) => match a_string.parse::<f64>() {
+                Ok(f) => Ok(f as i64),
+                _ => Err(ShellError::CantConvert(
+                    "into int".to_string(),
+                    "string".to_string(),
+                    span,
+                )),
+            },
         },
     }
 }
 
 #[cfg(test)]
 mod test {
+    use super::Value;
     use super::*;
+    use nu_protocol::Type::Error;
 
     #[test]
     fn test_examples() {
         use crate::test_examples;
 
         test_examples(SubCommand {})
     }
+
+    #[test]
+    fn turns_to_integer() {
+        let word = Value::test_string("10");
+        let expected = Value::test_int(10);
+
+        let actual = action(&word, Span::unknown(), 10);
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn turns_binary_to_integer() {
+        let s = Value::test_string("0b101");
+        let actual = action(&s, Span::unknown(), 10);
```

Comment:
I'm surprised this doesn't have to be a 2 instead of a 10. Do you know why it works, by chance?

Reply:
Good catch. When a string is matched, we check if the string starts with `0b`. If yes, then the "given" radix is ignored and 2 is used instead. Same thing for a string starting with "0x", we default to 16. You can see this in line 188 and line 216. 

I think the code can be slightly improved.